src/torrent/utils/thread.h

```diff
@@ -9,6 +9,8 @@
 #include <sys/types.h>
 
 #include <torrent/common.h>
+#include <torrent/net/resolver.h>
+#include <torrent/poll.h>
```

Comment:
Try removing the `nullptr` from `m_poll{nullptr};` and the other, those headers should not be included here.

Reply:
That indeed fixed it.